docs/dev/guidelines/client.rst

```diff
@@ -35,11 +42,43 @@ Some general aspects:
 
 1. Do not export types/functions unless you need to share it across multiple components.
 2. Do not introduce new types/values to the global namespace.
-3. Shared types should be defined in 'types.ts'.
+3. Shared types/interfaces should be defined in 'types.ts'.
 4. Within a file, type definitions should come first.
-5. Interfaces and types offer almost the same functionality. To ensure consistency, choose ``interface`` over ``type`` whenever possible.
+5. Interfaces and types offer almost the same functionality. To ensure consistency, choose ``interface`` over ``type`` whenever possible
+
+.. code-block:: ts
+
+    // Dont do
+    type AngularLink = {
+        text: string;
+        routerLink: (string | number) [];
+    }
+
+    // Do
+    interface AngularLink {
+        text: string;
+        routerLink: (string | number) [];
+    }
+
+    // And this is also allowed (because interface is not possible here)
+    type RouterLinkPart = string | number;
```

Comment:
This should be indented once to be part of the list item rather than being its own top-level block.

I.e.,
```rst
5. Interfaces and types offer almost the same functionality. To ensure consistency, choose ``interface`` over ``type`` whenever possible

    .. code-block:: ts
        // Dont do
        type AngularLink = {
        // …
```

Same for the other code block below.

Reply:
Done, please have another look.